File: network/src/main/java/com/app/network/ResponseCallBack.java
package com.app.network;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class ResponseCallBack<T> implements Callback<T> {
  @Override
  public void onResponse(final Call<T> call, final Response<T> response) {

  }

  @Override
  public void onFailure(final Call<T> call, final Throwable t) {

  }
}

File: network/src/main/java/com/app/network/ResponseCallBack.kt
package com.app.network

import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class ResponseCallBack<T> constructor(private val delegate: ResponeDelegate<T>) : Callback<T> {

    override fun onResponse(call: Call<T>, response: Response<T>) {
        assertResponseNull(response);
        delegate.onResponse(call, response);
    }

    override fun onFailure(call: Call<T>, t: Throwable) {
        delegate.onError(call, null);
    }

    fun assertResponseNull(respone: Response<T>?) {
        assert(respone != null);
    }
}
